### ISSUE TITLE
Getting Really Bad Results

### ISSUE THREAD

**Issue by Raneem-MT** [HuiqianLi/ASPS#8]
Hi, I am running your model to do polyp segmentation, my dataset has 4000 images, and I trained on batch size 8 for 25000 iterations, I tried doing inference on different iterations but the results seem quite bad (miou no more than 0.5), and the masks are very grainy. 
I used efficient_sam_vitt (efficient_sam_vitt.pt) from the efficientsam link you added, I used it for training and inference. 
Am I doing something wrong?
Thank you.

**Comment by HuiqianLi** [comment]
I apologize for any inconvenience this may have caused you. What about the miou results of other models? Is your training data loaded correctly? (If other models work well, you can consider other methods.)

The edges of the masks might appear slightly jagged, but the accuracy should not be unreasonably low. I don't know what the results you got are. Have you double-checked that the model path in both efficient_sam/build_efficient_sam.py and option.py (for the mascan model) is correctly loaded? Alternatively, you could try substituting the efficient_sam model with the official one to see if that makes a difference.

**Comment by Raneem-MT** [comment]
Thank you for the prompt reply. Yes, I compared with other methods and they performed well compared to ASPS. I was confused by the following when running your method:
- The checkpoints in your readme have EfficientSAM-ASPS, I downloaded it but it is a pth file, but when I went to the EfficientSAM repo it had the pt file, which I used. I tried using the pth checkpoint you provided but it did not work.
- I used your vis.py to iterate through multiple datasets and save the masks, but I set the encoder to efficient sam here:
if args.sam == "vit_b":
        model_encoder = sam_model_registry["vit_b"](checkpoint="pretrained/sam_vit_b_01ec64.pth")
    else:
        model_encoder = sam_model_registry["vit_h"](checkpoint="pretrained/sam_vit_h_4b8939.pth")
    model_encoder = model_e
because that is what I used for training, as in the following was kept the same:
 parser.add_argument('--sam', type=str, default='efficient_sam_vitt', help='vit_b or vit_h or efficient_sam_vitt')
and so in training:
elif args.sam == "efficient_sam_vitt":
            SAM_model = build_efficient_sam_vitt()

So, I used this for inference:
model = Network(args).cuda()
checkpoint = torch.load(checkpoint_path)
model.load_state_dict(checkpoint['model_state_dict'])
model.eval()
    
SAM_model = build_efficient_sam_vitt().cuda()
SAM_model.load_state_dict(checkpoint['sam_model_state_dict'], strict=False)
SAM_model.eval()

Please let me know if I did something wrong, and what checkpoints to use for training and inference. 

Thank you.

**Comment by HuiqianLi** [comment]
1. You should use the checkpoint in the EfficientSAM repo. The model EfficientSAM-ASPS in my link is what I had trained in the polyp datasets.

2. In `vis.py`, I haven't written the EfficientSAM code, but you can use the code in `infer.py` for inference and visualization, in the commented-out # visualize section. Have you modified the `vis.py` code at this part: `model_encoder = build_efficient_sam_vitt()`? It should be correct.

3. The code for training and inference is correct.

4. Everything you mentioned should be fine. I was wondering if the mascan model in `option.py` has been loaded correctly? If everything is okay, then it might be that it just isn't effective. I dont know the reason(｡╯︵╰｡)
